example/snippets.cpp

```diff
@@ -1530,6 +1601,637 @@ void section_connection_pool(string_view server_hostname, string_view username,
     }
 }
 
+static std::string get_name() { return "John"; }
+
+#ifndef BOOST_NO_CXX17_HDR_OPTIONAL
+//[sql_formatting_incremental_fn
+// Compose an update query that sets first_name, last_name, or both
+std::string compose_update_query(
+    boost::mysql::format_options opts,
+    std::int64_t employee_id,
+    std::optional<std::string> new_first_name,
+    std::optional<std::string> new_last_name
+)
+{
+    // There should be at least one update
+    assert(new_first_name || new_last_name);
+
+    // format_context will accumulate the query as we compose it
+    boost::mysql::format_context ctx(opts);
+
+    // append_raw adds raw SQL to the generated query, without quoting or escaping.
+    // You can only pass strings known at compile-time to append_raw,
+    // unless you use the runtime function.
+    ctx.append_raw("UPDATE employee SET ");
+
+    if (new_first_name)
+    {
+        // format_sql_to expands a format string and appends the result
+        // to a format context. This way, we can build our query in small pieces
+        // Add the first_name update clause
+        boost::mysql::format_sql_to(ctx, "first_name = {}", *new_first_name);
+    }
+    if (new_last_name)
+    {
+        if (new_first_name)
+            ctx.append_raw(", ");
+
+        // Add the last_name update clause
+        boost::mysql::format_sql_to(ctx, "last_name = {}", *new_last_name);
+    }
+
+    // Add the where clause
+    boost::mysql::format_sql_to(ctx, " WHERE id = {}", employee_id);
+
+    // Retrieve the generated query string
+    return std::move(ctx).get().value();
+}
+//]
+#endif
+
+//[sql_formatting_formatter_specialization
+// We want to add formatting support for employee_t
+struct employee_t
```

Comment:
you might want to be consistent with _t suffixes on types. 
I know its difficult to be consistent. 
there is a similar employee struct in batch_insert.cpp which doesn't have the "_t" suffix

Reply:
Caused by https://github.com/boostorg/mysql/issues/237 - will fix when solving the issue.